plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/rules/resthandler/RestIndexCorrelationRuleAction.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.correlation.rules.resthandler;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.plugin.correlation.EventsCorrelationPlugin;
+import org.opensearch.plugin.correlation.rules.action.IndexCorrelationRuleAction;
+import org.opensearch.plugin.correlation.rules.action.IndexCorrelationRuleRequest;
+import org.opensearch.plugin.correlation.rules.action.IndexCorrelationRuleResponse;
+import org.opensearch.plugin.correlation.rules.model.CorrelationRule;
+import org.opensearch.plugin.correlation.utils.CorrelationConstants;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.rest.action.RestResponseListener;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Rest action for indexing correlation rules.
+ *
+ * @opensearch.api
+ */
+public class RestIndexCorrelationRuleAction extends BaseRestHandler {
+
+    private static final Logger log = LogManager.getLogger(RestIndexCorrelationRuleAction.class);
+
+    /**
+     * Default constructor
+     */
+    public RestIndexCorrelationRuleAction() {}
+
+    @Override
+    public String getName() {
+        return "index_correlation_rule_action";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(
+            new Route(RestRequest.Method.POST, EventsCorrelationPlugin.CORRELATION_RULES_BASE_URI),
```

Comment:
From the RFC it looks like users can only create rules through the global `_correlation` endpoint. 

```
POST /_correlation/rules

Request Body: 
{
  "correlate": [
    {
      "index": "vpc_flow",
      "query": "dstaddr:4.5.6.7 or dstaddr:4.5.6.6"
    },
    {
      "index": "windows",
      "query": "winlog.event_data.SubjectDomainName:NTAUTHORI*"
    },
    {
      "index": "ad_logs",
      "query": "ResultType:50126"
    },
    {
      "index": "app_logs",
      "query": "endpoint:/customer_records.txt"
    }
  ]
}
```

Would it also make sense to enable users to do this on a per index basis? 


```
POST /my_ndex/_correlation/rule/{$rule_id}

where {$rule_id} is the (optional) existing rule
```

If so, we can do this in a follow up PR.

Reply:
will address this in follow-up pr.

Reply:
Maybe open an issue for this?

Reply:
created the issue: https://github.com/opensearch-project/OpenSearch/issues/7184